### ISSUE TITLE
[Bug] Installing from main branch fails with Python 3.10

### ISSUE THREAD

**Issue by saitcakmak** [pytorch/botorch#1288]
`pip install --upgrade git+https://github.com/pytorch/botorch.git` fails with Python 3.10. `pip install botorch` still works. The issue seems to happen while building the wheel.

** Stack trace/error message **
```
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error


    pkg_resources.DistributionNotFound: The 'importlib-metadata; python_version < "3.8"' distribution was not found and is required by the application
```

cc @Balandat, @esantorella

**Comment by esantorella** [comment]
I have the same issue. Installation with Python 3.9 does work, as does `pip install botorch`. Here's a longer stack trace: https://gist.github.com/esantorella/7dc851ebc8b93a6ee0c1de41861ff8f3

I'm using a Conda environment with packages installed as in the "Contributing" section of https://github.com/facebook/Ax.

**Comment by Balandat** [comment]
I believe this has to do with the fact that the use of `setuptools_scm` in `setup()` of `setup.py` has been deprecated, and that no `setuptools` version available for 3.10 supports this anymore. #1292 should resolve this issue.

**Comment by Balandat** [comment]
Fixed by https://github.com/pytorch/botorch/pull/1292
